geopolars/src/geoseries.rs

```diff
@@ -640,6 +647,35 @@ impl GeoSeries for Series {
         }
     }
 
+    fn distance(&self, other: &Series) -> Result<Series> {
+        use geo::algorithm::EuclideanDistance;
+
+        let mut output_array = MutablePrimitiveArray::<f64>::with_capacity(self.len());
+
+        for (g1, g2) in iter_geom(self).zip(iter_geom(other)) {
```

Comment:
Curious whether we should have an assertion before this that both series have the same length. Thoughts?

Reply:
Good idea. Not sure how to pass gracefully exception to python.

Reply:
Asserts are used currently only in tests. I can add it here but i guess it will fail spectacularly in python